pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -872,8 +871,14 @@ func (s *BlockBuilderScheduler) finalizeObservations() {
 		}
 
 		// Sort observations by start offset
-		sort.Slice(observations, func(i, j int) bool {
-			return observations[i].spec.StartOffset < observations[j].spec.StartOffset
+		slices.SortFunc(observations, func(a, b *observation) int {
+			if a.spec.StartOffset < b.spec.StartOffset {
+				return -1
+			}
+			if a.spec.StartOffset > b.spec.StartOffset {
+				return 1
+			}
+			return 0
 		})
```

Comment:
```suggestion
		slices.SortFunc(observations, func(a, b *observation) int {
			return int(a.spec.StartOffset - b.spec.StartOffset)
})
```

Reply:
Can't we just use `cmp.Compare` here @colega, as @bboreham suggested?

Reply:
Discussed with @colega offline, will apply his suggestions.